### PR TITLE
jmx.py: Add authentication support

### PR DESCRIPTION
**Pull request by seqizz**

### CODE REVIEW

File: src/jmx.py [hunk 1]
```diff
@@ -14,6 +14,8 @@
 def parse_args():
     parser = ArgumentParser()
     parser.add_argument('--prefix', default='jmx')
+    parser.add_argument('--username', default='')
+    parser.add_argument('--password', default='')
```
Comment (hasegeli) [inline src/jmx.py]: You do not need the `default`s.  They would be none by default.  `--` means the argument is optional.

Comment (hasegeli) [reply src/jmx.py]: There are 6 scripts using with `--user`.  There are 2 scripts with `--pass`, 3 with `--password`.  I would certainly name the former as `--user`, and consider renaming the later.

File: src/jmx.py [hunk 2]
```diff
@@ -26,6 +28,20 @@ def main():
...
 
+    if args.username and args.password:
+        auth_list = [
+                    '--user',
+                    args.username,
+                    '--pass',
+                    args.password
+                    ]
+    elif ((not args.username and args.password)
+            or (args.username and not args.password)):
```
Comment (hasegeli) [inline src/jmx.py]: `elif bool(args.username) != bool(args.password)`

File: src/jmx.py (hunk 1 above)
Comment (seqizz) [reply src/jmx.py]: I didn't get the last comment. This is just for jmx plugin, so searching `plugin => 'jmx',` usage in puppet manifests didn't give me something else then godkings/warlords couple.
Maybe I am missing something but, are you recommending renaming `username-password` to  `user-pass` here?

**Pull request merged**